packages/components/src/components/post-accordion-item/post-accordion-item.tsx

```diff
@@ -0,0 +1,90 @@
+import { Component, Element, h, Host, Listen, Method, Prop, State, Watch } from '@stencil/core';
+import { version } from '../../../package.json';
+import { checkEmptyOrOneOf } from '../../utils';
+import { HEADING_LEVELS, HeadingLevel } from './heading-levels';
+
+@Component({
+  tag: 'post-accordion-item',
+  styleUrl: 'post-accordion-item.scss',
+  shadow: true,
+})
+export class PostAccordionItem {
+  private collapsible: HTMLPostCollapsibleElement;
+
+  @Element() host: HTMLPostAccordionItemElement;
+
+  @State() id: string;
+  @State() isOpen: boolean;
+  @State() headingTag: string;
+
+  /**
+   * If `true`, the element is initially collapsed otherwise it is displayed.
+   */
+  @Prop() readonly collapsed?: boolean = false;
+
+  /**
+   * Defines the hierarchical level of the accordion item header within the headings structure.
+   */
+  @Prop() readonly headingLevel?: HeadingLevel = 2;
+
+  @Watch('headingLevel')
+  validateHeadingLevel(newValue = this.headingLevel) {
+    checkEmptyOrOneOf(
+      newValue,
+      HEADING_LEVELS,
+      'The `headingLevel` property of the `post-accordion-item` must be a number between 1 and 6.',
+    );
+  }
+
+  connectedCallback() {
+    this.validateHeadingLevel();
+  }
+
+  componentWillLoad() {
+    this.isOpen = !this.collapsed;
+  }
+
+  componentWillRender() {
+    this.id = this.host.id || `a${crypto.randomUUID()}`;
```

Comment:
Can you move this to the componentWillLoad? It evaluates on every re-render.

https://stenciljs.com/docs/component-lifecycle